webui/react/src/pages/ResourcePool/ResourcepoolDetail.tsx

```diff
@@ -62,32 +63,70 @@ const ResourcepoolDetailInner: React.FC = () => {
   const { canManageResourcePoolBindings } = usePermissions();
   const agents = Loadable.getOrElse([], useObservable(clusterStore.agents));
   const resourcePools = useObservable(clusterStore.resourcePools);
+  const navigate = useNavigate();
+  const [canceler] = useState(new AbortController());
+
+  const [tabKey, setTabKey] = useState<TabType>(tab ?? DEFAULT_POOL_TAB_KEY);
+  const [poolsStats, setPoolsStats] = useState<V1RPQueueStat[]>();
+
+  const MAX_USABLE_NODES = 1_000;
+  const MAX_USABLE_SLOTS = 10_000;
+
   const pool = useMemo(() => {
     if (!Loadable.isLoaded(resourcePools)) return;
 
     return resourcePools.data.find((pool) => pool.name === poolname);
   }, [poolname, resourcePools]);
 
+  const totalSlots = useMemo(() => {
+    const total = agents
+      .map((agent) => {
+        let totalSlots = 0;
+        for (const val of Object.values(agent.slotStats.typeStats ?? {})) {
+          totalSlots += val.total;
+        }
+        return totalSlots;
+      })
```

Comment:
perhaps?
```suggestion
      .map((agent) =>
        Object.values(agent.slotStats.typeStats ?? {}).reduce((totalSlots, { total }) => totalSlots + total, 0)
      )
```

Reply:
ah forgot to refactor code here. 

i end up doing this. i think i can reduce 1 loop with this

```ts
const total = agents.reduce(
      (totalVal, { slotStats }) =>
        totalVal +
        Object.values(slotStats.typeStats ?? {}).reduce(
          (localTotal, { total }) => localTotal + total,
          0,
        ),
      0,
    );
```